examples/global/main.rs

```diff
@@ -8,13 +8,14 @@ use std::error::Error;
 use std::sync::RwLock;
 
 lazy_static! {
-    static ref SETTINGS: RwLock<Config> = RwLock::new(Config::default());
+    static ref SETTINGS: RwLock<Config> = RwLock::new({
+        Config::builder()
+            .set_default("property", 42).unwrap()
+            .build().unwrap()
+    });
 }
 
 fn try_main() -> Result<(), Box<dyn Error>> {
-    // Set property
-    SETTINGS.write()?.set("property", 42)?;
-
```

Comment:
i don't think these two versions are equivalent. the existing example is *mutable* global configuration, the second is *immutable* global configuration

Reply:
So, AFAICS, my change is equivalent, because `Config::set` sets a default internally, which is now done via `ConfigBuilder::set_default()`. Mutability is still provided, as the resulting `Config` object is still in a `RwLock`.

Reply:
Though, mutability on the `Config` object is not needed anymore, because there are no (non-deprecated) methods that take `&mut self`!

Reply:
Are you sure that's correct? Looks to me like `Config::set` sets an override, not a default. Thus it can be used for modifying the config *after* it's been parsed (as is done in the example).

Without using deprecated methods, it's not possible to modify the `Config` after its construction, since as you say, there are no methods that take `&mut self`. That also makes the RwLock redundant.

As far as I can tell, before the deprecation it was possible to modify global configuration programatically after its construction, and after deprecating the methods it is not. It could well be that this was intentional, but it *does* mean that some use-cases (including one this example demonstrates) are no longer possible

Reply:
Hm. Yes, you're right. Lets see what @szarykott says, but I guess we should re-think the deprecation (as you said)! :+1: